sysstat-post-process

```diff
@@ -504,7 +504,52 @@ for my $log_file (sort readdir($dh)) {
             exit;
             }
         }
-    }
+    } elsif ($log_file =~ /^iostat-stdout.txt$/) {
+	my $log_fh = $log_file;
+        open(TMP,"$log_fh") || die "Could not open $log_fh\n";
+        my $separator = $/;
+        $/ = '';
+        my $ymd_timestamp_ms; # Epochtime in milliseconds
+
+        while (my($timestamp, $header, @devicelines) = split('\n', <TMP>)) {
+            foreach my $device_entry (@devicelines) {
+		my %iosample;
+                my $iodev = (split ' ', ${device_entry})[0];
+                (my $unused,
```

Comment:
What is `$unused` capturing?  Can you put a comment here with an example line that is being parsed like we do elsewhere in this script?

Reply:
I initially wrote this parser using records, and I could fill an entire record hash in one read.  BUT, crucible uses strict, which means I have to pre-declare every variable before I use it, which in turn means I can't do some perl-ish things like reading in multiple fields of a hash/record at once because the hash key value of $iodev is undefined when the structure is populated and in strict mode that forces an error.  So, instead, I have to read the value of that hash key first, which I do in line 517.  

That brings us to '$unused'.  Since I was forced to pre-read the IO device names ($iodev above), I already had a variable populated, so I had three choices:

- Read by consuming the value, which adds extra work modifying the $device_entry line for no real reason.
- Read the IO device name into $iodev again (free from the split) but I think this would make the code less clear.
- Read the IO device name into a variable indicating we aren't using it (also free from the split).

The lines look like this (and I can add an example of output to the post-processor, it was just kind of large).

```
2021-05-26T14:47:49-0400
Device            r/s     w/s     rkB/s     wkB/s   rrqm/s   wrqm/s  %rrqm  %wrqm r_await w_await aqu-sz rareq-sz wareq-sz  svctm  %util
sdb              0.00    0.00      0.00      0.00     0.00     0.00   0.00   0.00    0.00    0.00   0.00     0.00     0.00   0.00   0.00
sdb1             0.00    0.00      0.00      0.00     0.00     0.00   0.00   0.00    0.00    0.00   0.00     0.00     0.00   0.00   0.00
sdc              0.00    0.00      0.00      0.00     0.00     0.00   0.00   0.00    0.00    0.00   0.00     0.00     0.00   0.00   0.00
sda             23.33    0.00    330.67      0.00     0.00     0.00   0.00   0.00    2.27    0.00   0.05    14.17     0.00   0.94   2.20
sda1             4.33    0.00     57.33      0.00     0.00     0.00   0.00   0.00    2.92    0.00   0.01    13.23     0.00   1.15   0.50
sda2            19.00    0.00    273.33      0.00     0.00     0.00   0.00   0.00    2.12    0.00   0.04    14.39     0.00   0.89   1.70
rhel_worker01-root   19.00    0.00    273.33      0.00     0.00     0.00   0.00   0.00    2.18    0.00   0.04    14.39     0.00   0.89   1.70
rhel_worker01-swap    0.00    0.00      0.00      0.00     0.00     0.00   0.00   0.00    0.00    0.00   0.00     0.00     0.00   0.00   0.00
rhel_worker01-home    0.00    0.00      0.00      0.00     0.00     0.00   0.00   0.00    0.00    0.00   0.00     0.00     0.00   0.00   0.00
```

Reply:
I would just add the first 3 lines of that example as a comment...we don't need to see a super detailed example, just something that makes it apparent what is being parsed in the simplest way possible